hack/inspect_pods.sh

```
@@ -0,0 +1,124 @@
+    echo "--------------------------------------------------"
+    continue # Skip to the next pod in the loop
+  fi
+  # ---
+
+  # 2. For OpenShift, check the ImageStream to find the original public/external
+  #    registry image.
+  # We parse the ImageStream name and tag from the pod's image field.
+  IMAGE_STREAM_NAME=$(echo "$IMAGE_STREAM_TAG" | cut -d':' -f1)
+  TAG_NAME=$(echo "$IMAGE_STREAM_TAG" | cut -d':' -f2)
+  if [[ -z "$TAG_NAME" ]]; then TAG_NAME="latest"; fi
+  EXTERNAL_IMAGE_URL=$(oc get is "$IMAGE_STREAM_NAME" -n "$NAMESPACE" -o=jsonpath="{.status.tags[?(@.tag=='$TAG_NAME')].items[0].dockerImageReference}" 2>/dev/null)
```

Shellcheck (0.11.0)

[warning] EXTERNAL_IMAGE_URL appears unused. Verify use (or export if used externally).

(SC2034)